src/backup_restore.cpp

```diff
@@ -153,10 +153,190 @@ void BackupAndRestore::backupAndRestoreIpzVpd(types::IPZVpdMap& io_srcVpdMap,
+            l_srcBinaryValue = types::BinaryVector(l_srcStrValue.begin(),
+                                                   l_srcStrValue.end());
+        }
+        else
+        {
+            // Read keyword value from DBus
+            // TODO: Uncomment when readKeyword API implemenation is done,
+            // call the API with required parameters.
+            /* l_srcBinaryValue = utils::readKeyword(i_srcPath,
+            * std::tuple<types::Record, types::Keyword>(l_srcRecordName,
+            l_srcKeywordName)); l_srcStrValue =
+            std::string(l_srcBinaryValue.begin(),
+            l_srcBinaryValue.end());
+            */
+        }
+
+        types::BinaryVector l_dstBinaryValue;
+        std::string l_dstStrValue;
+        if (!io_dstVpdMap.empty())
+        {
+            genericUtility::getKwVal(io_dstVpdMap.at(l_dstRecordName),
+                                     l_dstKeywordName, l_dstStrValue);
+            l_dstBinaryValue = types::BinaryVector(l_dstStrValue.begin(),
+                                                   l_dstStrValue.end());
+        }
+        else
+        {
+            // Read keyword value from DBus
+            // TODO: Uncomment when readKeyword API implemenation is done,
+            // call the API with required parameters.
+            /* l_dstBinaryValue = utils::readKeyword(i_dstPath,
+            * std::tuple<types::Record, types::Keyword>(l_dstRecordName,
+            l_dstKeywordName)); l_dstStrValue =
+            std::string(l_dstBinaryValue.begin(),
+            l_dstBinaryValue.end());
+            */
+        }
+
+        if (l_srcBinaryValue != l_dstBinaryValue)
+        {
+            // ToDo: Handle if there is no valid default value in the backup and
+            // restore config JSON.
+            if (l_dstBinaryValue == l_defaultBinaryValue)
+            {
+                // TODO: Write on hardware,
+                // Write on Dbus,
+                // To keep the data in sync between hardware and parsed map
+                // updating the io_dstVpdMap. This should only be done if write
+                // on hardware returns success. utils::writeKeyword(i_dstPath,
+                // types::IpzData(l_dstRecordName, l_dstKeywordName
+                // l_srcBinaryValue));
+
+                if (!io_dstVpdMap.empty())
+                {
+                    io_dstVpdMap[l_dstRecordName][l_dstKeywordName] =
+                        l_srcStrValue;
+                }
+                continue;
+            }
+
+            if (l_srcBinaryValue == l_defaultBinaryValue)
+            {
+                // TODO: Write on hardware,
+                // Write on Dbus.
+                // utils::writeKeyword(i_srcPath,
+                // types::IpzData(l_srcRecordName, l_srcKeywordName,
+                // l_dstBinaryValue));
+                if (!io_srcVpdMap.empty())
+                {
+                    io_srcVpdMap[l_srcRecordName][l_srcKeywordName] =
```

Comment:
is this there in the map ?

Reply:
yes, if map is not empty, it should contain this record and keyword.
throwing an exception if not found.

Reply:
could you point me to it